core/src/test/scala/org/apache/spark/serializer/SerializerSuite.scala

```diff
@@ -0,0 +1,50 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.apache.spark.serializer
+
+import java.io.EOFException
+
+import scala.reflect.ClassTag
+
+import org.scalatest.{Matchers, FunSuite}
+
+class SerializerSuite extends FunSuite with Matchers {
+  test("DeserializationStream closes input at end") {
+    val in = new IntDeserializationStream(2)
```

Comment:
this test passed before any changes were made.  I just wanted to double check this for my own sake, and figured I might as well include it an extra tests.